src/category_theory/opposites.lean

```diff
@@ -273,6 +273,43 @@ protected definition unop (α : F.op ≅ G.op) : G ≅ F :=
 
 end nat_iso
 
+namespace equivalence
+
+variables {D : Type u₂} [category.{v₂} D]
+
+/--
+An equivalence between categories gives an equivalence between the opposite categories.
+-/
+@[simps]
+def op (e : C ≌ D) : Cᵒᵖ ≌ Dᵒᵖ :=
+{ functor := e.functor.op,
+  inverse := e.inverse.op,
+  unit_iso := (nat_iso.op e.unit_iso).symm,
+  counit_iso := (nat_iso.op e.counit_iso).symm,
+  functor_unit_iso_comp' := λ X, by { apply has_hom.hom.unop_inj, dsimp, simp, }, }
+
+/--
+An equivalence between opposite categories gives an equivalence between the original categories.
+-/
+@[simps]
+def unop (e : Cᵒᵖ ≌ Dᵒᵖ) : C ≌ D :=
+{ functor := e.functor.unop,
+  inverse := e.inverse.unop,
+  unit_iso := nat_iso.of_components (λ X,
```

Comment:
Shouldn't this be just as short as the `def` above it? Is `nat_iso.unop` missing?

Reply:
Err... there's some good reason why it can't exist (Lean, quite reasonably, can't unify something) but I forget the details. I'll try to reconstruct them and write a comment.

Reply:
Oh, I see. No, `nat_iso.unop` already exists, and works fine. The problem is that here we would be trying to `unop` something like `e.functor.op ⋙ e.inverse.op`. That is equal to `(e.inverse ⋙ e.functor).op`, but not equal enough for Lean to allow using `nat_iso.unop`.

Reply:
No, I just wasn't trying hard enough. It's fixed now.